vercel-ignore.sh

```diff
@@ -4,14 +4,15 @@
 # and which to ignore
 
 echo "VERCEL_GIT_COMMIT_REF: $VERCEL_GIT_COMMIT_REF"
+echo "VERCEL_GIT_COMMIT_SHA: $VERCEL_GIT_COMMIT_SHA"
 
-if [[ "$VERCEL_GIT_COMMIT_REF" == "dev" || "$VERCEL_GIT_COMMIT_REF" == "staging" || "$VERCEL_GIT_COMMIT_REF" == "main"  ]] ; then
-  # Proceed with the build
-    echo "✅ - Build can proceed"
+if [[ "$VERCEL_GIT_COMMIT_REF" == "dev" || "$VERCEL_GIT_COMMIT_REF" == "staging" || "$VERCEL_GIT_COMMIT_REF" == "main" ]] ; then
+  echo "✅ - Build can proceed"
+  exit 1;
+elif [[ "$VERCEL_GIT_COMMIT_SHA" == "$(git rev-parse --short main)" ]] ; then
```

Comment:
I've extracted this to a function - if only my googling for bash syntax didn't fail me it should be OK (I've also tried to test this as best as I could and it seemed to work so 🤞 ). I've also improved this to check the hash of a branch on the remote since Vercel didn't really have appropriate refs fetched locally for the previous test to work properly